kedro-datasets/setup.py

```diff
@@ -36,9 +36,8 @@ def _collect_requirements(requires):
     "pandas.GBQQueryDataSet": [PANDAS, "pandas-gbq>=0.12.0, <0.18.0"],
     "pandas.HDFDataSet": [
         PANDAS,
-        "tables~=3.6.0; platform_system == 'Windows'",
-        "tables~=3.6, <3.9; platform_system != 'Windows' and python_version<'3.9'",
-        "tables~=3.6; platform_system != 'Windows' and python_version>='3.9'",
+        "tables~=3.6, <3.9; python_version<'3.9'",
+        "tables~=3.6; python_version>='3.9'",
```

Comment:
kedro-datasets now supports >= 3.9

```suggestion
        "tables~=3.6",
```